### PR TITLE
DET-1578: windows probe kills tests after 45

### PR DESCRIPTION
**Pull request by packplusplus**
Pids were waiting forever. This may kill tests but not children of children of tests. It seems like a fair trade off. Also, some consistency fixes for var names.

### CODE REVIEW

File: shell/probe/raindrop.ps1 [hunk 1]
```diff
@@ -1,14 +1,24 @@
 [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
 
 function Execute {
-    Param([String]$File)
+    Param([String]$file)
```
Comment (keitharobertson) [inline shell/probe/raindrop.ps1:4]: https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/strongly-encouraged-development-guidelines?view=powershell-7.4#use-pascal-case-for-parameter-names

I don't make the rules 🤷

File: shell/probe/raindrop.ps1 [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
+        $proc = Start-Process -FilePath $file -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile
+
+        $timeouted = $null
+        $proc | Wait-Process -Timeout 60 -ErrorAction SilentlyContinue -ErrorVariable timeouted
+
+        if ($timeouted)
+        {
+            $proc | kill
+            return 102
+        }
```
Comment (keitharobertson) [inline shell/probe/raindrop.ps1:18]: nit everywhere else uses this style 
```suggestion
        if ($timeouted) {
            $proc | kill
            return 102
        }
```

**Review by keitharobertson — APPROVED** [review]

File: shell/probe/raindrop.ps1 [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
 
     try {
         $stdoutTempFile = New-Item -path "$dir\stdout.log" -Force
         $stderrTempFile = New-Item -path "$dir\stderr.log" -Force
-        $R = (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile).ExitCode
-        $Code = if (Test-Path $File) {$R} Else {127}
-        return $Code
+        $proc = Start-Process -FilePath $file -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile
+
+        $timeouted = $null
```
Comment (makk94) [inline shell/probe/raindrop.ps1:11]: just b/c this is public code... should we choose a real word ha

File: shell/probe/raindrop.ps1 [hunk 4]
```diff
@@ -32,7 +42,7 @@ function FromEnv { param ([string]$envVar, [string]$default)
 }
 
 $ca = FromEnv "PRELUDE_CA" "prelude-account-us1-us-east-2.s3.amazonaws.com"
-$dir = FromEnv "PRELUDE_DIR" $ca
+$dir = FromEnv "PRELUDE_DIR" ".vst"
```
Comment (makk94) [inline shell/probe/raindrop.ps1:45]: the other diff is that nocturnal doesn't even check for an env variable PRELUDE_DIR. do we care about aligning there too?

File: shell/probe/raindrop.ps1 [hunk 5]
```diff
@@ -6,9 +6,18 @@ function Execute {
     try {
         $stdoutTempFile = New-Item -path "$dir\stdout.log" -Force
         $stderrTempFile = New-Item -path "$dir\stderr.log" -Force
-        $R = (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile).ExitCode
-        $Code = if (Test-Path $File) {$R} Else {127}
-        return $Code
+        $proc = Start-Process -FilePath $File -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile
+
+        $errVar = $null
```
Comment (makk94) [inline shell/probe/raindrop.ps1:11]: nit: `timeoutErr` is clearer

File: shell/probe/raindrop.ps1 [hunk 6]
```diff
@@ -6,9 +6,18 @@ function Execute {
...
+        $proc | Wait-Process -Timeout 45 -ErrorAction SilentlyContinue -ErrorVariable errVar
+
+        if ($errVar) {
+            $proc | kill
```
Comment (makk94) [inline shell/probe/raindrop.ps1:15]: curious if you tested / know if the early kill here affects the redirected stdout/err at all?

**Review by keitharobertson — APPROVED** [review]

File: shell/probe/raindrop.ps1 (hunk 6 above)
Comment (packplusplus) [reply shell/probe/raindrop.ps1:15]: It's on my list to test, but I haven't tested it yet.

**Review by makk94 — APPROVED** [review]

File: shell/probe/raindrop.ps1 (hunk 5 above)
Comment (packplusplus) [reply shell/probe/raindrop.ps1:11]: I will acquiesce to that once we are sure you can't get messages that aren't timeouts.

**Pull request merged**